Fixed issue #10758 : write_full behavior in libradosstriper

sponce opened the pull request:
It was returning ENOENT when the file did not exists, while it should just have created it without complaining.

Signed-off-by: Sebastien Ponce <sebastien.ponce@cern.ch>

dachary commented on src/libradosstriper/RadosStriperImpl.cc:
```diff
@@ -316,7 +316,7 @@ int libradosstriper::RadosStriperImpl::write_full(const std::string& soid,
 {
   int rc = trunc(soid, 0);
-  if (rc) return rc;
+  if (rc && rc != -2) return rc; // ENOENT is obviously ok
```
-ENOENT instead of -2 ?

dachary commented:
Could you please update the backport pull requests so that they are cherry-pick -x from this one ? It probably is best to do it when this pull request is merged. The -x of cherry-pick will include a reference to the original pull request which is very helpful when figuring out where the backport comes from.

dachary commented:
Could you also include, in the commit message Fixes: #10758 ? That will be used by redmine to match the commit with the issue.

sponce commented:
Oups forgot the -x again... I'll fix that and backport the ENOENT.

Sebastien

On Thu, 2015-02-05 at 06:05 -0800, Loic Dachary wrote:
> Could you please update the backport pull requests so that they are
> cherry-pick -x from this one ? It probably is best to do it when this
> pull request is merged. The -x of cherry-pick will include a reference
> to the original pull request which is very helpful when figuring out
> where the backport comes from.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
>

dachary commented:
It would also be nice to have a minimal unit test for this to protect against future regressions.

sponce commented:
For this fix, I should be done :
  - I've added a unit test
  - I've amended the commit message with the issue number
  - I cherry-picked -x for the backports
  - I've fixed the code with ENOENT
On top, I've seen that your tools compiled it ok and ran the tests
successfully.

Cheers,

Sebastien

On Thu, 2015-02-05 at 06:12 -0800, Loic Dachary wrote:
> It would also be nice to have a minimal unit test for this to protect
> against future regressions.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
>

dachary commented:
@sponce one last detail : the **Fixes: #10758** needs to be in the body of the commit for redmine to notice. While you're at it the title should start with the component. I.e. something like:
<pre>
libradosstriper: fix write_full when ENOENT
</pre>
Otherwise looks great :-)

sponce commented:
Done. Thanks for all the comments.
